nautobot/docs/additional-features/caching.md

```diff
@@ -189,3 +189,177 @@ CELERY_RESULT_BACKEND_TRANSPORT_OPTIONS = CELERY_BROKER_TRANSPORT_OPTIONS
+    "ssl": True,
+    "ssl_cert_reqs": ssl.CERT_REQUIRED,
+    "ssl_ca_certs": "/opt/nautobot/redis/ca.crt",
+    "ssl_certfile": "/opt/nautobot/redis/tls.crt",
+    "ssl_keyfile": "/opt/nautobot/redis/tls.key",
+}
+```
+
+#### `celery` TLS Configuration
+
+To enable TLS for celery the following additional options are required:
+
+```python
+import ssl
+
+CELERY_REDIS_BACKEND_USE_SSL = {
+    "ssl_cert_reqs": ssl.CERT_REQUIRED,
+    "ssl_ca_certs": "/opt/nautobot/redis/ca.crt",
+    "ssl_certfile": "/opt/nautobot/redis/tls.crt",
+    "ssl_keyfile": "/opt/nautobot/redis/tls.key",
+}
+CELERY_BROKER_USE_SSL = CELERY_REDIS_BACKEND_USE_SSL
+```
+
+### With Sentinel
+
+#### `django-redis` Sentinel TLS Configuration
+
+To enable TLS with Sentinel the following options are required:
+
+```python
+import ssl
+
+REDIS_SSL_SETTINGS = {
+    "ssl_cert_reqs": ssl.CERT_REQUIRED,
+    "ssl_ca_certs": "/opt/nautobot/redis/ca.crt",
+    "ssl_certfile": "/opt/nautobot/redis/tls.crt",
+    "ssl_keyfile": "/opt/nautobot/redis/tls.key",
+}
+
+DJANGO_REDIS_CONNECTION_FACTORY = "django_redis.pool.SentinelConnectionFactory"
+CACHES = {
+    "default": {
+        "BACKEND": "django_redis.cache.RedisCache",
+        "LOCATION": "redis://nautobot/0",  # in this context 'nautobot' is the redis master/service name
+        "OPTIONS": {
+            "CLIENT_CLASS": "django_redis.client.SentinelClient",
+            "CONNECTION_POOL_CLASS": "redis.sentinel.SentinelConnectionPool",
+            "CONNECTION_POOL_KWARGS": {
+                "ssl": True,
+                **REDIS_SSL_SETTINGS,
+            },
+            "PASSWORD": "",
+            "SENTINEL_KWARGS": {
+                "password": "",  # likely the same password from above
+                "ssl": True,
+                **REDIS_SSL_SETTINGS,
+            },
+            "SENTINELS": [
+                ("mysentinel.redis.example.com", 26379),
+                ("othersentinel.redis.example.com", 26379),
+                ("thirdsentinel.redis.example.com", 26379)
+            ],
+        },
+    },
+}
+```
+
+#### `django-cacheops` Sentinel TLS Configuration
+
+To enable TLS with Sentinel the following options are required:
+
+```python
+import ssl
+
+REDIS_SSL_SETTINGS = {
+    "ssl_cert_reqs": ssl.CERT_REQUIRED,
+    "ssl_ca_certs": "/opt/nautobot/redis/ca.crt",
+    "ssl_certfile": "/opt/nautobot/redis/tls.crt",
+    "ssl_keyfile": "/opt/nautobot/redis/tls.key",
+}
+
+CACHEOPS_REDIS = False
+CACHEOPS_SENTINEL = {
+    "db": 1,
+    "locations": [
+        ("mysentinel.redis.example.com", 26379),
+        ("othersentinel.redis.example.com", 26379),
+        ("thirdsentinel.redis.example.com", 26379)
+    ],
+    "service_name": "nautobot",
+    "socket_timeout": 10,
+    "sentinel_kwargs": {
+        "password": "",
+        "ssl": True,
+        **REDIS_SSL_SETTINGS,
+    },
+    "password": "",
+    **REDIS_SSL_SETTINGS,
+    # Everything else is passed to `Sentinel()`
```

Comment:
I'm still not entirely satisfied with how much duplication there is here compared to the earlier `django-cacheops Sentinel Configuration` section. Do you think it would flow any better to combine the two (and do the same for the other added sections here), something like:

````
#### `django-cacheops` Sentinel Configuration

Notable settings:

* `locations`: List of tuples or tuple of tuples with each inner tuple containing the name or IP address
of the Redis server and port for each sentinel instance to connect to
* `service_name`: the master/service name in redis sentinel
* Additional parameters may be specified in the `CACHEOPS_SENTINEL` dictionary which are passed directly to Sentinel

!!! note
    `locations` for `django-cacheops` has a different meaning than the `LOCATION` value for `django-redis`

!!! warning
    [`CACHEOPS_REDIS`](#cacheops_redis) and [`CACHEOPS_SENTINEL`](#cacheops_sentinel) are mutually exclusive and will result in an error if both are set.

Example:

```python
# uncomment the below if using TLS with Redis:
# import ssl
# 
# REDIS_SSL_SETTINGS = {
#     "ssl_cert_reqs": ssl.CERT_REQUIRED,
#     "ssl_ca_certs": "/opt/nautobot/redis/ca.crt",
#     "ssl_certfile": "/opt/nautobot/redis/tls.crt",
#     "ssl_keyfile": "/opt/nautobot/redis/tls.key",
# }

CACHEOPS_REDIS = False
CACHEOPS_SENTINEL = {
    "db": 1,
    "locations": [
        ("mysentinel.redis.example.com", 26379),
        ("othersentinel.redis.example.com", 26379),
        ("thirdsentinel.redis.example.com", 26379)
    ],
    "service_name": "nautobot",
    "socket_timeout": 10,
    "sentinel_kwargs": {
        "password": "",
        # uncomment if using TLS with Redis
        # "ssl": True,
        # **REDIS_SSL_SETTINGS,
    },
    "password": "",
    # uncomment if using TLS with Redis
    # **REDIS_SSL_SETTINGS,

    # Everything else is passed to `Sentinel()`
}
```

For more details on how to configure Cacheops to use Redis Sentinel see the documentation for [Cacheops setup](https://github.com/Suor/django-cacheops#setup).
````

Reply:
If we do it this way should we move the non-sentinel stuff back to the required-settings and mentions TLS nested in the code similarly?